### PR TITLE
[Bug] Apply exp gains speed to animation

### LINKED ISSUE #3644: [Bug] exp gain skip is ignored
>  [Original Discord report](https://discord.com/channels/1125469663833370665/1274758406540099634)

## Describe the bug

The Exp-Gain `Skip` settings seems to be broken/not respected anymore

## Expected behavior

When settings `Exp Gain: Skip` it should skip the animation for exp-gain completely

## Screenshots / Videos

https://github.com/user-attachments/assets/9f7fca2c-b7f4-4e30-a6d0-01b3c7eb34f9

### PR DESCRIPTION
**Pull request by flx-sta**
# TODO:
Somehow the ExpGainSpeed setting is being reset somewhere

----

fixes #3644

## What are the changes the user will see?

The user will not have to wait for the exp-animation of the currently active PKM when set to `skip`

## Why am I making these changes?

Exp-Gains-Speed should have always affected the animation of the currently fighting PKM

## What are the changes from a developer perspective?

- Added a `ExpGainsSpeed` enum.
- Moved `GameManager` -> `game.startBattle()` into `ClassicModeHelper` => `game.classicMode.startBattle()`
    - ⚠️ **That's why it's showing 133 file changes!** 

## How to test the changes?

- Set the Exp-Gains-Speed in the settings to `Skip`
- Fight a wave and have a look at the animatio of the exp-gain : it won't play

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [ ] The PR is self-contained and cannot be split into smaller PRs?
    - Technically the change for `startBattle` to move into classicModeHelper could be a separate one.. but .. meh
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by CodeTappert — DISMISSED** [review]

**Review by xsn34kzx — DISMISSED** [review]

File: src/test/utils/helpers/settingsHelper.ts [hunk 1]
```diff
@@ -11,5 +12,19 @@ export class SettingsHelper extends GameManagerHelper {
...
+   * Set the EXP gains speed settings
+   * @param speed exp gains speed to set
+   */
+  expGainsSpeed(speed: ExpGainsSpeed) {
+    this.game.scene.expGainsSpeed = speed;
+    this.log(`exp-gains-speed set to ${ExpGainsSpeed[speed]} (=${speed})!`);
+  }
+
+  private log(...params: any[]) {
+    console.log("Settings:", ...params);
```
Comment (DayKev) [inline src/test/utils/helpers/settingsHelper.ts:28]:
```suggestion
    console.log("Settings: ", ...params);
```

File: src/test/utils/helpers/classicModeHelper.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { BattleStyle } from "#app/enums/battle-style";
 import { Species } from "#app/enums/species";
 import { GameModes, getGameMode } from "#app/game-mode";
-import overrides from "#app/overrides";
+import overrides from "#app/overrides.js";
```
Comment (DayKev) [inline src/test/utils/helpers/classicModeHelper.ts:4]:
```suggestion
import overrides from "#app/overrides";
```

File: src/test/ui/type-hints.test.ts [hunk 3]
```diff
@@ -65,7 +65,7 @@ describe("UI - Type Hints", () => {
   it("check status move color", async () => {
     game.override.enemySpecies(Species.FLORGES).moveset([Moves.GROWL]);
 
-    await game.startBattle([Species.RAYQUAZA]);
+    await game.classicMode.startBattle([Species.RAYQUAZA]);
```
Comment (DayKev) [inline src/test/ui/type-hints.test.ts:68]: If you're migrating all the existing uses of `startBattle()` then you might as well remove the old `startBattle()` function in `gameManager.ts` too.

Comment (flx-sta) [reply src/test/ui/type-hints.test.ts:68]: I could. But first I'd have to resolve the todo 😢

**Pull request closed without merging**